src/Sparrow/Compression/LZ4.cs

```diff
@@ -79,6 +79,35 @@ protected struct LZ4_stream_t_internal
             public uint initCheck;
         }
 
+        public long Encode64LongBuffer(
+            byte* input,
+            byte* output,
+            long inputLength,
+            long outputLength,
+            int acceleration = ACCELERATION_DEFAULT)
+        {
+            if (inputLength < int.MaxValue && outputLength < int.MaxValue)
```

Comment:
There is an error here, LZ4 can't actually handle 2GB size, it is something a bit less than that, and you need to take that into account.